src/new-components/avatar/avatar.tsx

```diff
@@ -28,20 +30,17 @@ const AVATAR_COLORS = [
 
 type AvatarSize = 'xxs' | 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' | 'xxxl'
 
-const AVATAR_SIZES: AvatarSize[] = ['xxs', 'xs', 's', 'm', 'l', 'xl', 'xxl', 'xxxl']
-
 type Props = {
     className?: string
     colorList?: string[]
-    size?: AvatarSize
+    size?: ResponsiveProp<AvatarSize>
     avatarUrl?: string
     user: { name?: string; email: string }
 }
 
 function Avatar({ user, avatarUrl, size = 'l', className, colorList = AVATAR_COLORS }: Props) {
```

Comment:
I'd take this opportunity to refactor it even further towards how the new design system components work or expose their options. Specifically, I'd recommend replacing the `className` prop with one called `exceptionallySetClassName`. This allows to give the signal that the components are meant to be self-styled, and that overriding their styles is a thing to only do exceptionally.

Reply:
I'd also even remove the `colorList` prop, as it goes against styles being self-owned. I checked and it does not seem to be used in Twist or Todoist.

Reply:
> I'd recommend replacing the className prop with one called exceptionallySetClassName.

I did think about that, but that would have also been a breaking change.

Reply:
I'll add a deprecated flag on classname so it's not such a breaking change.

Reply:
> I'd also even remove the `colorList` prop, as it goes against styles being self-owned. I checked and it does not seem to be used in Twist or Todoist.

What should the background colour be then if these are removed? Or do you just mean remove the param but keep the internal list?

Reply:
Deprecating the `className` is fine too. I assume that would be alongside adding the new `exceptionallySetClassName` as the recommended alternative to the deprecated prop.

As for the `colorList`, I was proposing to get rid of the prop only. The internal list stays.

Reply:
> I assume that would be alongside adding the new `exceptionallySetClassName` as the recommended alternative to the deprecated prop.

Yes, and I've added a note to the depcated tag to use `exceptionallySetClassName`.

> The internal list stays.

Okey doke 👍🏻

Reply:
Which, BTW, this is a change that would also breaking. So if you're trying to avoid all breaking changes (which is a good strategy after all), then you might as well not remove the prop, and simply set it as deprecated as well.

Reply:
Good point 👍🏻